MapboxNavigation/StyleManager.swift

```diff
@@ -0,0 +1,134 @@
+import UIKit
+import Solar
+
+protocol StyleManagerDelegate: class {
+    func locationFor(styleManager: StyleManager) -> CLLocation
+    func styleManager(_ styleManager: StyleManager, didApply style: Style)
+    func styleManagerDidRefreshAppearance(_ styleManager: StyleManager)
+}
+
+class StyleManager: NSObject {
+    
+    weak var delegate: StyleManagerDelegate!
+    
+    var currentStyleType: StyleType?
+    var styles = [Style]() { didSet { applyStyle() } }
+    var automaticallyAdjustsStyleForTimeOfDay = true {
+        didSet {
+            resumeNotifications()
+        }
+    }
+    
+    required init(_ delegate: StyleManagerDelegate) {
+        self.delegate = delegate
+        super.init()
+        resumeNotifications()
+    }
+    
+    deinit {
+        suspendNotifications()
+    }
+    
+    func resumeNotifications() {
+        suspendNotifications()
+        
+        guard automaticallyAdjustsStyleForTimeOfDay else { return }
+        
+        let location = delegate.locationFor(styleManager: self)
+        guard let solar = Solar(coordinate: location.coordinate),
+            let sunrise = solar.sunrise,
+            let sunset = solar.sunset else {
```

Comment:
We should `isNightTime()`  over too https://github.com/mapbox/mapbox-navigation-ios/pull/904/files#diff-ab5a908dc637e51fa78cf2be28d033ecL288

Reply:
Oh it is, nvm. But why do we create two solars?

Reply:
B/c the coordinate is immutable. Take a flight from LA to Montreal and we'd have a discrepancy of ~6 hours if we would reuse the solar object. In addition to using a new solar object, we should also listen for timezone changes via UIApplicationSignificantTimeChangeNotification.